doc/design/doi-generation-1.md

```diff
@@ -0,0 +1,41 @@
+# Publish process
+
+This process is predicated on a few tasks that are expected to receive valid input and assuming that only certain errors could happen that are out of our control.
+Dandisets have a new attribute added to track their validation status: `VALID`, `INVALID`, and `VALIDATING`.
+Whenever a dandiset is modified (metadata changed or files uploaded/removed), it is set to `VALIDATING` and an asynchronous task is triggered to run metadata validation on it. The task will update the state of the dandiset accordingly when it finishes.
+
+- publish enabled only if a dandiset is `VALID`. The button will be greyed out in the UI and the API endpoint will return 405.
+- on publish trigger:
+   - create the versioned release in the db (this prevents any asset/blob belonging to the release from being deleted)
+   - insert all the relevant metadata (version, doi, url, manifest location/checksum) into the metadata record of the new version 
+  - send this new metadata record to mint the doi (there should be no reason why this should fail unless the external resource fails. if it fails notify admins since this is likely a new bug)
+   - write out to s3 bucket (even on any datacite failure)
```

Comment:
```suggestion
   - write out to s3 bucket (even on any datacite failure)
      - add checksums for dandiset.yaml and assets.yaml which should be stored in a checksums.json in the published directory.
```

Reply:
I'm not clear on the value of checksumming the manifests. It doesn't provide any kind of security since it's stored in the same place as the content being checksummed. It doesn't really give any additional data integrity during the download process that HTTPS doesn't already give. Hypothetically it could safe guard against the API server uploading corrupted data, but this is not the best way to solve that problem.

Could you clarify what this is supposed to achieve?

Reply:
i'm not sure aws provides checksums for things. this would mostly be a way for any user to verify that the dandiset.yaml or the assets.yaml have not changed. if the aws header already provides this, great, otherwise it's just another file containing the checksums of the files. i guess the AWS e-tag would be sufficient for the moment.